src/mds/FSMap.h

```diff
@@ -188,6 +188,10 @@ class Filesystem
   void dump(ceph::Formatter *f) const;
   void print(std::ostream& out) const;
 
+  bool is_upgradeable() const {
+    return !mds_map.allows_standby_replay() && mds_map.get_num_in_mds() <= 1;
+  }
```

Comment:
This is a temporary restriction, as I see it. It's a stop-gap between turning off all MDS (except rank 0 of one file system, our current recommendation) and enabling a rolling upgrade without reducing max_mds.

> Why not just keep the intersection of MDS features as the filesystem compat set?

You mean instead of storing the CompatSet in the MDSMap? Because those features are sometimes used to indicate on-disk format changes (snapv2 for example).